lib/i2c.ex

```diff
@@ -217,7 +217,43 @@ defmodule Circuits.I2C do
 
     :"do not show this result in output"
   end
-
+  
+  @doc """
+  Given a list of potential devices, find the bus and address. 
+  
+  Since I2C devises often have several hard coded addresses 
+  and may be on any bus, it may be helpful to discover the device. 
+  
+  Fails if there is more than one potential match, or there are no matches.
```

Comment:
I'm torn on failing if there's more than one match since that could happen. It seems like the caller would know whether it wants only the first one or more than that. Plus, this function is a nice one-liner if it returns a list. 

Another option would be to call this `discover_first!` and return the first one found or raise. I haven't seen the code that calls this, but I'm figuring that it's something along the lines of if the user doesn't specify an I2C bus and address that the default is to find one for them.

Reply:
Sold. I will tweak it and add discover_one!.